src/jit/SimdArm32Inl.h

```diff
@@ -267,25 +590,161 @@ static void simdEmitI32x4Dot(sljit_compiler* compiler, sljit_s32 vd, sljit_s32 v
+    }
+}
+
+static void simdEmitF32x4Binary(sljit_compiler* compiler, JITArg src[2], JITArg dst, sljit_s32 op)
+{
+    for (uint8_t i = 0; i < 4; i++) {
+        sljit_emit_fmem(compiler, SLJIT_MOV_F32 | SLJIT_MEM_UNALIGNED | SLJIT_MEM_LOAD, SLJIT_FR0, src[0].arg, src[0].argw + (i * sizeof(float)));
+        sljit_emit_fmem(compiler, SLJIT_MOV_F32 | SLJIT_MEM_UNALIGNED | SLJIT_MEM_LOAD, SLJIT_FR1, src[1].arg, src[1].argw + (i * sizeof(float)));
+        sljit_emit_fop2(compiler, op, dst.arg, dst.argw + (i * sizeof(float)), SLJIT_FR0, 0, SLJIT_FR1, 0);
+    }
+}
+
+static void simdEmitFloatUnaryOpWithCB(sljit_compiler* compiler, JITArg src, JITArg dst, unaryCallbackFunction cb)
+{
+    if (src.arg == SLJIT_MEM1(kFrameReg)) {
+        sljit_emit_op2(compiler, SLJIT_ADD, SLJIT_R0, 0, kFrameReg, 0, SLJIT_IMM, src.argw);
+    } else {
+        ASSERT(src.arg == SLJIT_MEM0());
+        sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_R0, 0, SLJIT_IMM, src.argw);
+    }
+
+    sljit_emit_op2(compiler, SLJIT_ADD, SLJIT_R1, 0, kFrameReg, 0, SLJIT_IMM, dst.argw);
+    sljit_emit_icall(compiler, SLJIT_CALL, SLJIT_ARGS2(VOID, P, P), SLJIT_IMM, GET_FUNC_ADDR(sljit_sw, cb));
+}
+
+static void simdEmitFloatBinaryOpWithCB(sljit_compiler* compiler, JITArg src[2], JITArg dst, binaryCallbackFunction cb)
+{
+    if (src[0].arg == SLJIT_MEM1(kFrameReg)) {
+        sljit_emit_op2(compiler, SLJIT_ADD, SLJIT_R0, 0, kFrameReg, 0, SLJIT_IMM, src[0].argw);
+    } else {
+        ASSERT(src[0].arg == SLJIT_MEM0());
+        sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_R0, 0, SLJIT_IMM, src[0].argw);
+    }
+
+    if (src[1].arg == SLJIT_MEM1(kFrameReg)) {
+        sljit_emit_op2(compiler, SLJIT_ADD, SLJIT_R1, 0, kFrameReg, 0, SLJIT_IMM, src[1].argw);
+    } else {
+        ASSERT(src[1].arg == SLJIT_MEM0());
+        sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_R1, 0, SLJIT_IMM, src[1].argw);
+    }
+
+    sljit_emit_op2(compiler, SLJIT_ADD, SLJIT_R2, 0, kFrameReg, 0, SLJIT_IMM, dst.argw);
+    sljit_emit_icall(compiler, SLJIT_CALL, SLJIT_ARGS3(VOID, P, P, P), SLJIT_IMM, GET_FUNC_ADDR(sljit_sw, cb));
+}
+
+static void simdEmitF32x4CMP(sljit_compiler* compiler, JITArg src[2], JITArg dst, sljit_s32 opcode, sljit_s32 flag)
+{
+    opcode |= SLJIT_CMP_F32;
+    auto tmpReg = SLJIT_R0;
+
+    for (uint8_t i = 0; i < 4; i++) {
+        sljit_emit_fmem(compiler, SLJIT_MOV_F32 | SLJIT_MEM_UNALIGNED | SLJIT_MEM_LOAD, SLJIT_FR0, src[0].arg, src[0].argw + (i * sizeof(float)));
+        sljit_emit_fmem(compiler, SLJIT_MOV_F32 | SLJIT_MEM_UNALIGNED | SLJIT_MEM_LOAD, SLJIT_FR1, src[1].arg, src[1].argw + (i * sizeof(float)));
+        sljit_emit_fop1(compiler, opcode, SLJIT_FR0, 0, SLJIT_FR1, 0);
+        sljit_emit_op_flags(compiler, SLJIT_MOV32, tmpReg, 0, flag);
+        sljit_emit_op2(compiler, SLJIT_SUB, tmpReg, 0, SLJIT_IMM, 0, tmpReg, 0);
+        sljit_emit_mem(compiler, SLJIT_MOV32 | SLJIT_MEM_UNALIGNED | SLJIT_MEM_STORE, tmpReg, dst.arg, dst.argw + (i * sizeof(float)));
```

Comment:
This looks like a lot of instructions. Perhaps a loop could be better or use a callback.